core/src/Kokkos_Core_fwd.hpp

```diff
@@ -184,6 +186,30 @@ using DefaultHostExecutionSpace KOKKOS_IMPL_DEFAULT_HOST_EXEC_SPACE_ANNOTATION =
     "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::OpenMP, Kokkos::Threads, Kokkos::Experimental::HPX, or Kokkos::Serial."
 #endif
 
+// check for devices that support sharedSpace
+#if defined(KOKKOS_ENABLE_CUDA)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION = CudaUVMSpace;
+#define KOKKOS_HAS_SHARED_SPACE 1
+#elif defined(KOKKOS_ENABLE_HIP)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION = HIPManagedSpace;
+#define KOKKOS_HAS_SHARED_SPACE 1
+#elif defined(KOKKOS_ENABLE_SYCL)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION =
+    Experimental::SYCLSharedUSMSpace;
+#define KOKKOS_HAS_SHARED_SPACE 1
+// if only host compile point to HostSpace
+#elif !defined(KOKKOS_ENABLE_OPENACC) && !defined(KOKKOS_ENABLE_OPENMPTARGET)
```

Comment:
This exclusive phrasing feels risky, since anyone doing a future backend addition will have to find and adjust this. It would be preferable if there were a definition somewhere else that described "backend with distinct memory from `HostSpace` is enabled", and then we could check the absence of that.

Reply:
yeah, I totally agree, I don't like it ether. I like your suggestion though. We should probably hand the definition to the backends ...

Reply:
We can't do much that is future proof here and that is ok.  We have the same issue with our defining default execution spaces.